librz/cons/bar.c

```diff
@@ -0,0 +1,81 @@
+// SPDX-FileCopyrightText: 2022 Anton Kochkov <anton.kochkov@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_cons.h>
+#include <rz_util/rz_assert.h>
+
+// TODO: add support for colors
+/**
+ * \brief Create the string buffer with the progressbar
+ *
+ * \param opts Progressbar options: color, style, legend
+ * \param pc How much percent is filled
+ * \param width Width of the histogram
+ */
+RZ_API RZ_OWN RzStrBuf *rz_progressbar(RZ_NONNULL RzBarOptions *opts, int pc, int width) {
+	rz_return_val_if_fail(opts, NULL);
+	RzStrBuf *buf = rz_strbuf_new("");
+	if (!buf) {
+		return NULL;
+	}
+	int i, cols = (width == -1) ? 78 : width;
+	const char *h_line = opts->unicode ? RUNE_LONG_LINE_HORIZ : "-";
+	const char *block = opts->unicode ? UTF_BLOCK : "#";
+
+	pc = RZ_MAX(0, RZ_MIN(100, pc));
+	if (opts->legend) {
+		rz_strbuf_appendf(buf, "%4d%% ", pc);
+	}
+	cols -= 15;
+	rz_strbuf_append(buf, "[");
+	for (i = cols * pc / 100; i; i--) {
+		rz_strbuf_append(buf, block);
+	}
+	for (i = cols - (cols * pc / 100); i; i--) {
```

Comment:
`i` can be defined in the loop.

Reply:
It's reused in two different loops, I don't think it's really beneficial.

Reply:
yes, but you can redefine it so it gets optimized by the compiler.